packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx

```diff
@@ -56,20 +64,29 @@ export const AddNewProposalModal = () => {
   const feeInfo = useTransactionFee(member?.controllerAccount, transaction)
 
   useEffect((): any => {
-    if (!state.matches('requirementsVerification')) {
-      return
-    }
-
-    if (!member) {
-      return showModal<SwitchMemberModalCall>({ modal: 'SwitchMember' })
-    }
-
-    if (feeInfo && feeInfo.canAfford) {
-      return send('NEXT')
+    if (state.matches('requirementsVerification')) {
+      if (!member) {
+        return showModal<SwitchMemberModalCall>({ modal: 'SwitchMember' })
+      }
+
+      if (feeInfo && feeInfo.canAfford) {
+        return send('NEXT')
+      }
+
+      if (feeInfo && !feeInfo.canAfford) {
+        return send('FAIL')
+      }
     }
 
-    if (feeInfo && !feeInfo.canAfford) {
-      return send('FAIL')
+    if (state.matches('generalParameters.stakingAccount') && !hasRequiredStake) {
```

Comment:
So, the main benefit of using a state machine is to have multiple `boolean` values turned into distinct states. I think that we can model this state into the machine.

There should be another step between `selectType` and `generalDetails` in which we check the required stake, it can transition to either:

*   `generalDetails` or
*   `requiredStakeFail` (final state) -> in which we display the "required stake" modals